src/node_sqlite.cc

```diff
@@ -441,7 +441,9 @@ void StatementSync::All(const FunctionCallbackInfo<Value>& args) {
 
     for (int i = 0; i < num_cols; ++i) {
       Local<Value> key = stmt->ColumnNameToValue(i);
+      if (key.IsEmpty()) return;
       Local<Value> val = stmt->ColumnToValue(i);
+      if (val.IsEmpty()) return;
```

Comment:
nit... is it worth combining these into a single `if (key.IsEmpty() || val.IsEmpty()) return` just to make things a bit more compact?

Reply:
We usually fail fast when a JavaScript exception is pending to avoid unsafe calls into JavaScript in that state. It seems that `ColumnToValue()` does not call into JavaScript, but unless someone feels strongly, I'd err on the side of caution here.